application_sandbox/dynamic_rendering_cube/main.cpp

```diff
@@ -0,0 +1,318 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "application_sandbox/sample_application_framework/sample_application.h"
+#include "support/entry/entry.h"
+#include "vulkan_helpers/buffer_frame_data.h"
+#include "vulkan_helpers/helper_functions.h"
+#include "vulkan_helpers/vulkan_application.h"
+#include "vulkan_helpers/vulkan_model.h"
+
+#include <chrono>
+#include "mathfu/matrix.h"
+#include "mathfu/vector.h"
+
+using Mat44 = mathfu::Matrix<float, 4, 4>;
+using Vector4 = mathfu::Vector<float, 4>;
+
+namespace cube_model {
+#include "cube.obj.h"
+}
+const auto& cube_data = cube_model::model;
+
+uint32_t cube_vertex_shader[] =
+#include "cube.vert.spv"
+    ;
+
+uint32_t cube_fragment_shader[] =
+#include "cube.frag.spv"
+    ;
+
+struct CubeFrameData {
+  containers::unique_ptr<vulkan::VkCommandBuffer> command_buffer_;
+  containers::unique_ptr<vulkan::VkFramebuffer> framebuffer_;
+  containers::unique_ptr<vulkan::DescriptorSet> cube_descriptor_set_;
+};
+
+VkPhysicalDeviceDynamicRenderingFeaturesKHR kDynamicRenderingFeat{
+    VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_DYNAMIC_RENDERING_FEATURES_KHR,
+    nullptr,
+    VK_TRUE
+};
+
+// This creates an application with 16MB of image memory, and defaults
+// for host, and device buffer sizes.
+class CubeSample : public sample_application::Sample<CubeFrameData> {
+ public:
+  CubeSample(const entry::EntryData* data)
+      : data_(data),
+        Sample<CubeFrameData>(
+            data->allocator(), data, 1, 512, 1, 1,
+            sample_application::SampleOptions()
+                .EnableMultisampling()
+                .AddDeviceExtensionStructure(&kDynamicRenderingFeat),
+            {0}, {}, {VK_KHR_DYNAMIC_RENDERING_EXTENSION_NAME}),
+        cube_(data->allocator(), data->logger(), cube_data) {}
+  virtual void InitializeApplicationData(
+      vulkan::VkCommandBuffer* initialization_buffer,
```

Comment:
nit: it would be nicer if we could call `vkGetPhysicalDeviceFeatures2` with the VkPhysicalDeviceDynamicRenderingFeatures structure included in the pNext chain of the VkPhysicalDeviceFeatures2 to check if the feature is supported or not.

Reply:
Done.